benchmarks/dbscan/print_timers.cpp

```diff
@@ -25,38 +28,31 @@ struct Timer
   double duration;
 };
 
-std::vector<Timer> arborx_dbscan_example_timers;
+std::stack<Timer> current_timers;
+std::vector<Timer> done_timers;
 
 } // namespace
 
-void arborx_dbscan_example_set_create_profile_section(char const *label,
-                                                      std::uint32_t *id)
-{
-  *id = arborx_dbscan_example_timers.size();
-  arborx_dbscan_example_timers.push_back({label, {}, {}});
-}
-
-void arborx_dbscan_example_set_destroy_profile_section(std::uint32_t) {}
-
-void arborx_dbscan_example_set_start_profile_section(std::uint32_t id)
+void ArborX_Benchmark::push_region(char const *label)
 {
   Kokkos::fence();
   auto now = Timer::clock_type::now();
-  auto &timer = arborx_dbscan_example_timers[id];
-  timer.tick = now;
+  current_timers.push({label, now, {}});
 }
 
-void arborx_dbscan_example_set_stop_profile_section(std::uint32_t id)
+void ArborX_Benchmark::pop_region()
 {
   Kokkos::fence();
   auto now = Timer::clock_type::now();
-  auto &timer = arborx_dbscan_example_timers[id];
+  auto timer = current_timers.top();
+  current_timers.pop();
   timer.duration = std::chrono::duration<double>(now - timer.tick).count();
+  done_timers.push_back(timer);
 }
```

Comment:
I don't think this is a correct structure, though I don't know if you make an assumption of uniqueness of the names in the timers that we want to get. You lose the notion of nested when you push into `done_timers`. So you two timers with the same name, like sorting, but nested differently, would appear in multiple places in flat array without a way to distinguish them.

I think we would need some hierarchical data structure. Every level of it is `std::set` (so that the timers get merged), and every level has a parent. I'm not sure what's the easiest way to do that.

Reply:
We agreed we would ignore the nested structure and only look at the actual region label.  If we change our mind, we can just traverse the stack as we pop a region.